src/download.c

```diff
@@ -420,11 +419,16 @@ static int perform_curl_io_and_complete(int *left)
 		curl_multi_remove_handle(mcurl, handle);
 		curl_easy_cleanup(handle);
 		file->curl = NULL;
-	}
 
-	curlm_ret = curl_multi_perform(mcurl, left);
-	if (curlm_ret != CURLM_OK) {
-		return -1;
+		/* "bounded" is false when the remainder of the multi-stack is
+		 * to be processed, ignoring the hysteresis bound. */
+		if (bounded) {
+			left--;
+		}
```

Comment:
I added a docstring for the function that describes the "numfds" parameter, and "left" is initialized to its value. Maybe I could decrement "numfds" instead. Would that be more clear?